### PR TITLE
feat: add `multiple` prop to the `InternalDropdown` component

### PR DESCRIPTION
**Pull request by HellWolf93**
fix: #1596

## Changes proposed in this PR:
- Add `multiple` prop to InternalDropdown component

- [X] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

### CODE REVIEW

File: src/components/InternalDropdown/index.js [hunk 1]
```diff
@@ -208,15 +231,35 @@ const InternalDropdown = forwardRef((props, reference) => {
...
+                    }
+                    return onChange(value.concat([option]));
+                }
+                if (value) {
+                    return onChange([value, option]);
+                }
+                return onChange([option]);
+            }
+            return onChange(option);
+        },
```
Comment (reiniergs) [inline src/components/InternalDropdown/index.js:250]: this function and the one above are kinds of the same, can we reuse it somehow?

File: src/components/InternalDropdown/helpers/getValueNames.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export default function getValueNames(value) {
+    if (Array.isArray(value)) {
+        const names = value.map(v => v.name);
+        return names.filter(v => !!v);
```
Comment (reiniergs) [inline src/components/InternalDropdown/helpers/getValueNames.js:4]: I that the options are going to be few and this code will run lightning-fast, but :) ... you are iterating twice ... you can use reduce here and make 1 iteration e.g.

```js
return value.reduce((seed, item) => {
    if (item.name) {
        seed.push(name);
    }
    return seed;
}, []);
```

**Comment by reiniergs** [comment]
@HellWolf93 I don't see test for the component, are you gonna write integration tests for multiple selection??

/cc @TahimiLeonBravo @LeandroTorresSicilia

File: src/components/InternalDropdown/helpers/getValueNames.js (hunk 2 above)
Comment (HellWolf93) [reply src/components/InternalDropdown/helpers/getValueNames.js:4]: you are right, thanks.

**Pull request merged**